src/basic/env-util.c

```diff
@@ -452,6 +452,18 @@ char *strv_env_get(char **l, const char *name) {
         return strv_env_get_n(l, name, strlen(name), 0);
 }
 
+char *strv_env_pairs_get(char **l, const char *name) {
+        char **key, **value, *result = NULL;
+
+        assert(name);
+
+        STRV_FOREACH_PAIR(key, value, l)
+                if (key && value && streq(*key, name))
```

Comment:
no need to check key/value for non-NULL, the STRV_FOREACH_PAIR macro already guarantees that